Tribler/community/tunnel/tunnel_community.py

```diff
@@ -643,6 +657,10 @@ def check_create(self, messages):
             if self.crypto.key.pub().key_to_bin() != message.payload.node_public_key:
                 yield DropMessage(message, "TunnelCommunity: public keys do not match")
                 continue
+            
+            if not self.settings.become_exitnode and message.payload.become_exit:
+                yield DropMessage(message, "Exit-node functionality disabled")
```

Comment:
I would reply with a message saying denied or something like that, then choose a different peer if you receive a denied response. Otherwise creating a tunnel might be ridicously difficult.

Moreover, how are you going to link two tunnels? Are introduction/rendezvous points exit points? Because they do exit_data.